packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1454,7 +1454,7 @@ class _TabBarViewState extends State<TabBarView> {
       _pageController.jumpToPage(_currentIndex!);
       _warpUnderwayCount -= 1;
     }
-    if (widget.children != oldWidget.children && _warpUnderwayCount == 0) {
+    if (widget.children != oldWidget.children) {
```

Comment:
I appreciate that removing the warpUnderway check doesn't cause any tests to fail.  I'm not sure if I agree with the rationale for landing this change being "maybe it will not break any internal tests".

The warpUnderway count brackets statements like:
```dart
await _pageController.animateToPage(_currentIndex!, duration: duration, curve: Curves.ease);
```
To really test this change, we'd need to prove that it's OK to remove the tab being moved ("warped"), or all the tabs, or all the tabs to the left or right of the tab being moved, etc.

I'm the one who included this check ([#7387](https://github.com/flutter/flutter/pull/7387). It's been 5 1/2 years, and so I have to admit that I can't remember exactly what motivated it. Sorry about that; I should have a left a comment behind.

Reply:
@HansMuller 
Many thanks for your insight. Now that I have a better understanding of the inner workings of TabBarView, I agree that removing this check is not a good option. Mainly because it changes one of the design choices for this widget which is to defer children update until the transition ends. 
When the transition ends the following code takes care of updating children (if needed).
https://github.com/flutter/flutter/blob/774e4526181797fae60fa770cbb98b2e658551d8/packages/flutter/lib/src/material/tabs.dart#L1543-L1550 


Looking more carefully at the related issue (https://github.com/flutter/flutter/issues/107399), the problem is that children are not updated after a transition to an **_adjacent_** tab. They are correctly updated after a transition to a non adjacent tab (because of the code mentioned above). 

I have open another PR (https://github.com/flutter/flutter/pull/112168) and close this one as the description of the solution is very different and it will be easier for future contributors to understand why this PR was a dead end.